gen/generate.go

```diff
@@ -234,7 +243,9 @@ func generateModule(m *compile.Module, i thriftPackageImporter, builder *generat
 		}
 
 		// TODO(abg): Verify no file collisions
-		files["constants.go"] = buff.Bytes()
+		if !o.NoConstants {
+			files["constants.go"] = buff.Bytes()
```

Comment:
I didn't want to do this because if you do this for services (below), the GenerateServiceRequest will not be populated properly. builder.AddRootService has to be called, so I have to do that logic regardless, and then I did the same logic here to be consistent and in case there are side effects from functions called inside the if statement (I think in the case of types and constants, there aren't, but this needs to be guaranteed by the documentation of functions called).